### PR TITLE
Use the built-in library for time zone functions

### PR DESCRIPTION
**Pull request by mborsetti**
## Proposed change

Use the built-in library for timezone functions.

Benefits:
* Implemented best practice of using built-in libraries when possible;
* Improved speed of line 120 of ``chile.py`` by ~1.6x;
* Removed dependency on external package ``python-dateutil`` for timezone.

## Type of change

- [ ] New country holidays support (thank you!)
- [ ] Supported country holidays update (calendar discrepancy fix, localization)
- [x] Existing code/tests/processes improvement (best practice, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (adds functionality to python-holidays in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by PPsyrius** [comment]
As a minor note, Japan's "Vernal Equinox Day" (春分の日) got the same problem here so they could probably get the same treatment as the Chilean one here.

**Comment by mborsetti** [comment]
> As a minor note, Japan's "Vernal Equinox Day" (春分の日) got the same problem here so they could probably get the same treatment as the Chilean one here.

Indeed, not sure how I missed it, thanks!  PR update forthcoming.

File: holidays/countries/chile.py [hunk 1]
```diff
@@ -23,6 +22,12 @@
 from holidays.constants import TUE, WED, THU, FRI, SAT, SUN
 from holidays.holiday_base import HolidayBase
 
+# use standard library for timezone
+try:
+    from zoneinfo import ZoneInfo
+except ImportError:  # pragma: no cover
```
Comment (arkid15r) [inline holidays/countries/chile.py:28]: It's not a coverage improvement, it's a coverage decrease suppression :)

I had a similar problem in [this PR](https://github.com/dr-prodigy/python-holidays/pull/889/). It took me a while to figure out a better solution. I'll take care of the refactoring after #889 is merged.

**Review by arkid15r — COMMENTED** [review]
Great optimization! LGTM

**Comment by arkid15r** [comment]
Thanks for another great contribution Mike!

File: holidays/countries/chile.py (hunk 1 above)
Comment (mborsetti) [reply holidays/countries/chile.py:28]: Yeah, I agree, it's definitely a suppression.

I can't think how to do it "correctly", so I look forward to seeing what you come up with! :)

Comment (arkid15r) [reply holidays/countries/chile.py:28]: You can find an example in https://github.com/dr-prodigy/python-holidays/pull/889, I'm going to use pretty much the same approach.

**Pull request merged**